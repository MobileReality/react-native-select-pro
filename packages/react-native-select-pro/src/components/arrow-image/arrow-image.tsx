import React from 'react';
import type { ImageStyle } from 'react-native';
import { Animated, Image, StyleSheet } from 'react-native';

import { useSelectContext } from '../../context';
import { useAnimation } from '../../hooks';

const arrowImage = require('./../../assets/icons/chevron-down.png');

export const ArrowImage = () => {
    const { isOpened, styles: mainStyles, animation, arrowImageProp } = useSelectContext();
    const rotateAnimation = useAnimation({ isOpened, animation });
    const { arrow } = mainStyles?.select ?? {};

    if (rotateAnimation) {
        const rotate = rotateAnimation.interpolate({
            inputRange: [0, 1],
            outputRange: ['0deg', '180deg'],
        });

        return (
            <Animated.Image
                source={arrowImage}
                {...arrowImageProp}
                style={[styles.arrowIcon, { transform: [{ rotate }] }, arrow?.icon]}
            />
        );
    }

    return (
        <Image
            source={arrowImage}
            {...arrowImageProp}
            style={[styles.arrowIcon, isOpened && styles.arrowIconOpened, arrow?.icon]}
        />
    );
};

type Styles = {
    arrowIcon: ImageStyle;
    arrowIconOpened: ImageStyle;
};

const styles = StyleSheet.create<Styles>({
    arrowIcon: {
        width: 24,
        height: 24,
        zIndex: -1,
    },
    arrowIconOpened: {
        transform: [{ rotate: '180deg' }],
    },
});
