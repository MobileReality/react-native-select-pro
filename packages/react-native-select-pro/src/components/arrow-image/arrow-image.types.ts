import type { ComponentProps } from 'react';

import type { Arrow } from '../arrow';

type FromArrowComponentProps = Pick<
    ComponentProps<typeof Arrow>,
    'isOpened' | 'animation' | 'arrowIconStyles'
>;

export type ArrowImageProps = FromArrowComponentProps;
