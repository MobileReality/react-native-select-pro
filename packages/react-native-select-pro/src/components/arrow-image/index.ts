export { ArrowImage } from './arrow-image';
