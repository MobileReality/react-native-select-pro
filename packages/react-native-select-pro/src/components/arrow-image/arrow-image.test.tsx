import React from 'react';
import { render } from '@testing-library/react-native';

import { ArrowImage } from './arrow-image';

describe('ArrowImage', () => {
    it('should generate ArrowImage snapshot', () => {
        const arrowImage = render(<ArrowImage />);
        expect(arrowImage).toMatchSnapshot();
    });
});
