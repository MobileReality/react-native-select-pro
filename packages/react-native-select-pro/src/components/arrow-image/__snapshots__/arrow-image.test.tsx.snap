// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowImage should generate ArrowImage snapshot 1`] = `
<Image
  collapsable={false}
  source={
    {
      "testUri": "packages/react-native-select-pro/src/assets/icons/chevron-down.png",
    }
  }
  style={
    {
      "height": 24,
      "transform": [
        {
          "rotate": "0deg",
        },
      ],
      "width": 24,
      "zIndex": -1,
    }
  }
/>
`;
